File: swift/0066-plus-one.swift
class Solution {
    func plusOne(_ digits: [Int]) -> [Int] {
        // parameters are immutable in swift
        var digit = digits
        for i in stride(from: digit.count - 1, through: 0, by: -1) {
            digit[i] += 1
            
            if digit[i] <= 9 {
                return digit
            }
            
            digit[i] = 0
        }
        
        digit.insert(1, at: 0)
        return digit
    }
}

File: go/0066-plus-one.go
func plusOne(digits []int) []int {
    for i := len(digits) - 1; i >= 0; i-- {
        if digits[i] < 9 {
            digits[i] += 1
            return digits
        }
        digits[i] = 0
    }
    digits[0] = 1
    digits = append(digits, 0)
    return digits
}